README.md

```diff
@@ -83,21 +83,71 @@ sintax $(your_command). For instance:
 ```
 DATABASE=postgres://$(whoami)@localhost/database
 ```
- 
+
 Running it on the CLI:
 
 ```
 $ dotenv "echo $DATABASE"
 postgres://myusername@localhost/database
 ```
 
+### Type checking envs
+Env variables are simple strings. However, they can represent other types like
+integers, booleans, IP addresses, emails, URIs, and so on. We provide an interface
+that performs type checking after loading the envs and before running your application.
+If the type-check succeeded the application is executed, otherwise you will get an
+error with the types that mismatch.
+
+In order to use this functionality you can use the `loadSafeFile` which takes the same
+configuration value as the `loadFile` function. Also, you need to have a `.scheme.yml`
+in your current directory. This file must have the following structure:
+
+```yaml
+- type: bool
+  envs:
+    - name: DOTENV
+      required: true
+    - name: OTHERENV
+      required: false
+- type: integer
+  envs:
+    - name: PORT
+      required: true
+    - name: TOKEN
+      required: true
+```
+
+It is a list of type and envs. So, in this example, `DOTENV` must have a value
+of `true` or `false` otherwise it won't be parsed as a boolean value. And envs
+like `PORT` must be any integer. Currently, we are supporting the following types:
+
+- `bool` - Accepts values `false` or `true`
+- `integer` - Accepts values of possitive integers
+- `text` - Any text
+
+You must set **required** if the env var is going to be type checked or not.
```

Comment:
Perhaps it would be better to say something like this: 

`**require** specifies if the env var is obligatory or not. In case you set it to true but do not provide it, you wil get an exception.`

Perhaps this reflects better the implementation we discussed ... :)